File: pd.js
var threshold;

function setThreshold(value)
{
    threshold = value
}

function calculatePd(solvencyRatio)
{
    var pd;

    if(solvencyRatio < threshold)
    {
        pd = ["R1", 5]
    }
    else
    {
        pd = ["R2", 20]
    }

    console.log(pd);

    return pd;
}

setThreshold(20);
calculatePd(10)

File: pd.hs
import Data.Time.Clock

data PD = PD { rating :: String, probability :: Int } deriving Show

calculatePd :: Int -> PD
calculatePd solvencyRatio = if solvencyRatio < 20 then PD "R1" 5 else PD "R2" 20


displayPd :: PD -> IO()
displayPd pd = do
    time <- getCurrentTime
    putStrLn ("Rating is " ++ rating pd ++ " at " ++ show time)
